### PR TITLE
Email campaign support

### PR DESCRIPTION
**Pull request by DriftingSands**

### CODE REVIEW

File: blocks/email-composer/email-composer.js [hunk 1]
```diff
@@ -187,64 +208,54 @@ export default async function decorate(block) {
...
-        throw new Error(res.status);
-      })
-        .then(async (res) => {
-          project = res.project;
-
-          block.querySelector('.actions').innerHTML = `
+      block.querySelector('.actions').innerHTML = `
             <a href="#" target="_blank" class="button secondary action copy">Copy</a>
-            <a href="${project.driveUrl}" target="_blank" class="button action secondary edit">Edit</a>
+            <a href="${project?.driveUrl || `https://da.live/edit#/da-self-service/${id}/newsletter`}" target="_blank" class="button action secondary edit">Edit</a>
```
Comment (icaraps) [inline blocks/email-composer/email-composer.js:213]: I dont think you can hardcode newsletter here

**Review by icaraps — CHANGES_REQUESTED** [review]

File: blocks/email-composer/email-composer.js (hunk 1 above)
Comment (DriftingSands) [reply blocks/email-composer/email-composer.js]: I removed it

File: blocks/email-composer/email-composer.js [hunk 2]
```diff
@@ -16,6 +16,27 @@ export default async function decorate(block) {
...
+    if (!project) {
+      // try dark alley
+      const darkAlleyProjectResponse = await fetch(`${SCRIPT_API}/darkAlleyList/${id}`);
+      const darkAlleyProjectData = await darkAlleyProjectResponse.json();
+      project = darkAlleyProjectData?.project;
+    }
+
+    // no project
+    if (!project) {
+      block.querySelector('.content p').textContent = OOPS;
```
Comment (icaraps) [inline blocks/email-composer/email-composer.js:37]: Add a return here or the code will continue to execute

**Comment by icaraps** [comment]
Can you add the edit feature that takes you directly to the file in gdrive like you added it recently on the pages overview ?

File: blocks/email-composer/email-composer.js (hunk 2 above)
Comment (DriftingSands) [reply blocks/email-composer/email-composer.js:37]: good catch👍

File: styles/styles.css [hunk 3]
```diff
@@ -302,6 +302,27 @@ input[type="color"] {
   text-transform: none;
 }
 
+
+
+.button.edit.loading,
```
Comment (icaraps) [inline styles/styles.css:307]: Can you use a single classname instead of 2 ?

Comment (DriftingSands) [reply styles/styles.css]: yep

**Pull request merged**